commit 44b43162473d801eaeaf8810e13c514f83408218
Author: Trent Mick <trentm@gmail.com>
Date:   2019-05-23T15:25:22-07:00 (5 months ago)
    
    TRITON-1701 sdcadm.test.js test for 'sdcadm --version' doesn't handle branch names with dots
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/test/sdcadm.test.js b/test/sdcadm.test.js
index 2e2279d..868cfe3 100644
--- a/test/sdcadm.test.js
+++ b/test/sdcadm.test.js
@@ -9,8 +9,9 @@
  */
 
 
-var test = require('tape').test;
 var exec = require('child_process').exec;
+var format = require('util').format;
+var test = require('tape').test;
 
 
 test('sdcadm', function (t) {
@@ -41,7 +42,9 @@ test('sdcadm --help', function (t) {
 test('sdcadm --version', function (t) {
     exec('sdcadm --version', function (err, stdout, stderr) {
         t.ifError(err, 'no version error');
-        t.ok(stdout.match(/^sdcadm \d+\.\d+\.\d+ \([\w-]+-\d+T\d+Z-.+\)/));
+        var verRe = /^sdcadm \d+\.\d+\.\d+ \(.+-\d+T\d+Z-g[0-9a-f]{7}\)$/;
+        t.ok(verRe.test(stdout.trim()),
+            format('version should match %s: %j', verRe, stdout.trim()));
         t.equal(stderr, '');
 
         t.end();
